### ISSUE TITLE
Informative debug print statements during training

### ISSUE THREAD

**Issue by SietsmaRJ** [molgenis/capice#128]
**Is your feature request related to a problem? Please describe.**

- It is not related to a problem.
- During training (which takes a while) I would love to know if a newly added feature is added to the features that XGBoost can learn on. 
- I would love to know how many samples are going into the training as train samples and test samples. 
- Since we are going to support balancing again, I would love to know how many samples are used in training per consequence.

**Describe the solution you'd like**

- Debug level verbosity logging statement that describe the `self.processed_features` right before the `train` method is called.
- Debug level verbosity logging statement that describes the amount of samples within the training set and within the test set, again right before the `train` method is called.
- Method that extracts all unique consequences present within specifically the training set and counts the amount of rows that contain this consequence. Command to extract all unique consequences within data `x` (assuming the column names have been corrected of their `%` sign due to the BCFTools conversion): `pd.Series(x['Consequence'].str.split('&', expand=True).values.ravel()).dropna().sort_values(ignore_index=True).unique()` and to count the amount of rows (assuming `c` is your `for c in consequences`: `x[x['Consequence'].str.contains(c)].shape[0]`. Also DEBUG level. EDIT: this point can also be implemented within the balancing script itself. As long as it is documented somewhere, this makes validating a model performance easier as you now know how many training samples are present per consequence, so how well a model should perform for each and every consequence.

**Describe alternatives you've considered**

None.

**Additional context**
